scalafix-core/shared/src/main/scala/scalafix/syntax/package.scala

```diff
@@ -30,6 +31,10 @@ package object syntax {
     @deprecated("Renamed to denotation", "0.5.0")
     def denotOpt: Option[Denotation] = denotation
     def denotation: Option[Denotation] = index.denotation(symbol)
+    def resultType(dialect: Dialect): Option[Type] =
+      denotation.flatMap(denot =>
+        DenotationOps.resultType(symbol, denot, dialect))
+    def resultType: Option[Type] = resultType(DenotationOps.defaultDialect)
```

Comment:
Can we only keep this overload and let advanced users with custom dialects call `DenotationOps.resultType(customDialect)`?

Reply:
fine by me